src/plugins/usage_collection/server/collector/collector.ts

```diff
@@ -45,11 +45,18 @@ export type MakeSchemaFrom<Base> = {
     : RecursiveMakeSchemaFrom<Required<Base>[Key]>;
 };
 
+export interface CollectorFetchClients {
```

Comment:
When I first see the word "Context", I immediately think about "request context". Other options would be "CollectorFetchOptions" (but we already refer to 'options' elsewhere) or "collectorFetchProviders" but that's also not great. For now, I'd leave it as `CollectorFetchClients` and think about the right name when we add `KibanaRequest`.

Reply:
>  think it's best to define the name in this PR so we don't bother other teams only because we've renamed a TS interface (and, most likely, its mock)
That's a very valid point. Considering `fetch` is request driven, I'll go with your suggestion of `CollectorFetchContext`, mainly because I can't come up with a different name right now.